addon/ng2/utilities/change.ts

```diff
@@ -0,0 +1,103 @@
+'use strict';
+
+import * as Promise from 'ember-cli/lib/ext/promise';
+let fs = require('fs');
+
+const readFile =  Promise.denodeify(fs.readFile);
+const writeFile = Promise.denodeify(fs.writeFile);
+
+export interface Change {
+  /**
+   *  True on success, false otherwise.
+   */
+  apply(): Promise<void>;
+
+  // The file this change should be applied to. Some changes might not apply to
+  // a file (maybe the config).
+  path: string | null;
+
+  // The order this change should be applied. Normally the position inside the file.
+  // Changes are applied from the bottom of a file to the top.
+  order: number | null;
+
+  // The description of this change. This will be outputted in a dry or verbose run.
+  description: string;
+}
+
+
+/**
+ * Will add text to the source code.
+ */
+export class InsertChange implements Change {
+  const order: number;
+  const description: string;
+  constructor(
+    public path: string,
```

Comment:
Styling issue: please put the arguments of this function on the same line as the function declaration, if it exceeds 100 characters just cut the line there at the closest argument.